src/NuGet.Services.BasicSearch/SecretReader/SecretReaderFactory.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using NuGet.Indexing;
+using NuGet.Services.KeyVault;
+
+namespace NuGet.Services.BasicSearch
+{
+    public class SecretReaderFactory : ISecretReaderFactory
+    {
+        public const string VaultName = "keyVault:VaultName";
```

Comment:
I think it might be better to give these constants a name that implies that they are strings used to access values in the config rather than the values themselves, e.g. `public const string ConfigVaultName = "keyVault:VaultName";`